src/helper.c

```diff
@@ -58,6 +58,13 @@ int calculate_digest(const EVP_MD* md, const uint8_t* data, size_t len, uint8_t*
 
 int byte_array_init(ByteArray* arr, const uint8_t* data, int len)
 {
+    if (len == 0)
+    {
```

Comment:
Could you just change the brackets to be aligned with the .clang-format ( { after if condition) for consistency?